kuma/static/js/components/banners/utils/banners-event-util.js

```diff
@@ -0,0 +1,74 @@
+(function() {
+    'use strict';
+    var mdnBannerEvents = {
+        /**
+         * Attach common event handlers for the CTA banner.
+         * @param {Object} banner - The banner element
+         */
+        attachEvents: function(banner) {
+            this.handleCollapseEvent(banner);
+            this.handleCloseEvent(banner);
+            this.handleExternalLinkEvent(banner);
+        },
+        /**
+         * Handles click events on the close(x) icon
+         * @param {Object} banner - The banner element
+         */
+        handleCloseEvent: function(banner) {
+            var closeButton = banner.querySelector('.mdn-cta-close');
+
+            if (!closeButton) {
+                return;
+            }
+
+            closeButton.addEventListener('click', function() {
+                banner.classList.add('hidden');
+                window.mdnBannersStateUtil.setBannerInactive(
+                    banner.dataset['banner']
+                );
+            });
+        },
+        /**
+         * Handles click events on the collapse icon
+         * @param {Object} banner - The banner element
+         */
+        handleCollapseEvent: function(banner) {
+            var collapseButton = banner.querySelector('.mdn-cta-collapse');
+
+            if (!collapseButton) {
+                return;
+            }
+
+            collapseButton.addEventListener('click', function() {
+                banner.classList.remove('expanded');
+                banner.classList.add('is-collapsed');
+            });
+        },
+        /**
+         * Handles click events on elements(typically a button) with
+         * the class `.mdn-cta-external-link`
+         * @param {Object} banner - The banner element
+         */
+        handleExternalLinkEvent: function(banner) {
+            var externalLink = banner.querySelector('.mdn-cta-external-link');
+
+            if (!externalLink) {
+                return;
+            }
+
+            externalLink.addEventListener('click', function(event) {
+                event.preventDefault();
+                event.stopPropagation();
+                window.open(event.target.href);
```

Comment:
event.target might be the wrong element, if the link has a span or an image inside of it, for example. I think you can just use externalLink here instead of event.target. Or use event.currentTarget.

Instead of using window.open(), however (since it could get us in trouble wiht popup blockers, maybe) would it be okay to just code the link in the banner with `target="_blank" rel="noopener"` and then not have an event handler at all?

Reply:
Much better, will change.